torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1331,7 +1331,7 @@ def make_singular_matrix_batch_base(size, rank):
         u_diag_abs_largest = u_diag_abs.max(dim=-1, keepdim=True).values
         u_diag_abs_smallest_idxs = torch.topk(u_diag_abs, k=(n - rank), largest=False).indices
         u.diagonal(0, -2, -1).div_(u_diag_abs_largest)
-        u[..., u_diag_abs_smallest_idxs] = torch.finfo(dtype).eps
+        u.diagonal(0, -2, -1)[..., u_diag_abs_smallest_idxs] = torch.finfo(dtype).eps
```

Comment:
These changes were introduced in https://github.com/pytorch/pytorch/pull/66933, and the samples produced were wrong. CC @Lezcano

Reply:
They are wrong in what sense?

Reply:
Ah, I see now. For that to do what I wanted to do, it needed a `triu()`, yep.

Reply:
This change modified indexed columns, not diagonals.